### PR TITLE
FreeBSD: catch up with 1300124 version bump

### PR DESCRIPTION
**Pull request by mjguzik**
- use cache_vop_mkdir
- cache_rename -> cache_vop_rename

Signed-off-by: Mateusz Guzik <mjguzik@gmail.com>

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [ ] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (a change to man pages or other documentation)

### Checklist:

- [ ] My code follows the OpenZFS [code style requirements](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md#coding-conventions).
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**contributing** document](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md).
- [ ] I have added [tests](https://github.com/openzfs/zfs/tree/master/tests) to cover my changes.
- [ ] I have run the ZFS Test Suite with this change applied.
- [ ] All commit messages are properly formatted and contain [`Signed-off-by`](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md#signed-off-by).

### CODE REVIEW

**Comment by mjguzik** [comment]
Sorry about that, the ifdef covered FreeBSD 12, but not the older development branch this still builds against.

File: module/os/freebsd/zfs/zfs_vnops_os.c [hunk 1]
```diff
@@ -1826,7 +1836,7 @@ zfs_rmdir_(vnode_t *dvp, vnode_t *vp, const char *name, cred_t *cr)
 
 	dmu_tx_commit(tx);
 
-	cache_purge(vp);
+	cache_vop_rmdir(dvp, vp);
```
[the remark this answers is not in the document]
Comment (mjguzik) [reply module/os/freebsd/zfs/zfs_vnops_os.c:1837]: A variant is implemented above based on said version.

**Comment by freqlabs** [comment]
This builds against the stable/12 snapshot from yesterday, it shouldn't be that old :)

File: module/os/freebsd/zfs/zfs_vnops_os.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (allanjude) [reply module/os/freebsd/zfs/zfs_vnops_os.c:1837]: I misread, got it.

**Review by allanjude — APPROVED** [review]

**Review by behlendorf — APPROVED** [review]

**Review by freqlabs — APPROVED** [review]

**Comment by freqlabs** [comment]
Builds fine with the fix from #11139

**Pull request merged**